### PR TITLE
chore(eslint): migrate to v9.x

### PR DESCRIPTION
**Pull request by yin1999**
Migrate the configuration of eslint to v9.x. Check: https://eslint.org/docs/latest/use/migrate-to-9.0.0

Run `npm run lint` to check the migration.

### CODE REVIEW

**Comment by yin1999** [comment]
> Would it be fine to directly use https://github.com/windingwind/zotero-plugin-template/blob/main/eslint.config.mjs ?

How should I handle the `Zotero_Tabs` global variable. I'm not familiar with it.

File: src/modules/services/google.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ async function _google(url: string, data: Required<TranslateTask>) {
 
     let e, f, g;
 
+    // disable the eslint, as the code is copied from google translate
+    /* eslint-disable */
```
Comment (yin1999) [inline src/modules/services/google.ts:23]: This part is a bit hard to be modified. I think we can leave it unchanged.

File: src/elements/base.ts [hunk 2]
```diff
@@ -15,20 +15,19 @@ export class PluginCEBase extends XULElementBase {
     // Following the connectedCallback from XULElementBase
     let content: Node = this.content;
     if (content) {
-      content = document.importNode(content, true);
+      content = Zotero.getMainWindow().document.importNode(content, true);
```
[the remark this answers is not in the document]
Comment (yin1999) [reply src/elements/base.ts]: Should we also allow to use `window` directly for this folder?

[the remark this answers is not in the document]
Comment (windingwind) [reply src/elements/base.ts]: Yeah that would be ideal

[the remark this answers is not in the document]
Comment (yin1999) [reply src/elements/base.ts]: Done

File: eslint.config.mjs [hunk 3]
```diff
@@ -45,12 +45,11 @@ export default tseslint.config(
     },
   },
   {
-    // allow to use the `document` global directly in CEs
+    // allow to use the `document` and `window` globals directly in CEs
```
Comment (windingwind) [inline eslint.config.mjs:48]: Well, isn’t that also enables using window in all folders? Ideally should only be allowed for CEs.

Comment (yin1999) [reply eslint.config.mjs:48]: This is only a overriding config for `src/elements/*.ts`, which override the `no-restricted-globals` rule configured in  the previous config (for all the `.ts` files).

**Review by windingwind — COMMENTED** [review]
Thanks!

**Pull request merged**